Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -364,68 +358,71 @@ public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot
+                        if (this.options.VerificationToken != null && slackEvent.token != this.options.VerificationToken)
                         {
                             response.StatusCode = 403;
                         }
                         else
                         {
                             Activity activity = new Activity()
                             {
-                                Timestamp = new DateTime(),
+                                Timestamp = default(DateTime),
                                 ChannelId = "slack",
                                 Conversation = new ConversationAccount()
                                 {
-                                    Id = slackEvent.Channel.Id
+                                    Id = slackEvent.Channel.Id,
                                 },
                                 From = new ChannelAccount()
                                 {
-                                    Id = (slackEvent.BotId != null) ? slackEvent.BotId : slackEvent.User.Id
+                                    Id = (slackEvent.BotId != null) ? slackEvent.BotId : slackEvent.User.Id,
                                 },
                                 Recipient = new ChannelAccount()
                                 {
-                                    Id = null
+                                    Id = null,
                                 },
                                 ChannelData = slackEvent,
-                                Type = ActivityTypes.Event
+                                Type = ActivityTypes.Event,
                             };
 
                             // Extra fields that do not belong to activity
-                            activity.Conversation.Properties["thread_ts"] = slackEvent.ThreadTs;
+                            activity.Conversation.Properties["thread_ts"] = ((dynamic)slackEvent)["event"].thread_ts;
                             activity.Conversation.Properties["team"] = slackEvent.Team.Id;
 
                             // this complains because of extra fields in conversation
-                            activity.Recipient.Id = await GetBotUserByTeamAsync(activity);
+                            activity.Recipient.Id = await this.GetBotUserByTeamAsync(activity);
 
                             // create a conversation reference
                             using (var context = new TurnContext(this, activity))
                             {
                                 context.TurnState.Add("httpStatus", "200");
 
-                                await RunPipelineAsync(context, bot.OnTurnAsync, default(CancellationToken));
+                                try
+                                {
+                                    await this.RunPipelineAsync(context, bot.OnTurnAsync, default(CancellationToken));
+                                }
+                                catch (Exception ex)
```

Comment:
If we are only doing a `throw ex;` we can delete this try/catch block (the exception should be handled in one of the caller methods)